### PR TITLE
feat: #54 add filter by email on AllUsers page

### PR DESCRIPTION
**Pull request by Ekaterina1994**

### CODE REVIEW

File: mw-webapp/src/dataAccessLogic/UserPreviewDAL.ts [hunk 1]
```diff
@@ -13,15 +13,15 @@ export class UserPreviewDAL {
   /**
    * Get amount of all users in collection
    */
-  public static async getUsersPreviewAmount(): Promise<number> {
-    return await UserService.getUsersDTOAmount();
+  public static async getUsersPreviewAmount(filterEmail?: string): Promise<number> {
```
Comment (tritonJS826) [inline mw-webapp/src/dataAccessLogic/UserPreviewDAL.ts:16]: use open close pronciple to write right arguments
(filter: {email?: string})

File: mw-webapp/src/logic/allUsersPage/AllUsersPage.tsx [hunk 2]
```diff
@@ -1,17 +1,30 @@
...
 import styles from "src/logic/allUsersPage/AllUsersPage.module.scss";
 
+const FILTER_EMAIL_EMPTY_STRING = "";
+const DEFAULT_ALL_USERS_PAGE_SETTINGS: AllUsersPageSettings = {filterEmail: FILTER_EMAIL_EMPTY_STRING};
+
+/**
+ * Safe opened tab from localStorage
+ */
+const allUsersPageSettingsValidator = (currentSettings: AllUsersPageSettings) => {
+  return !!currentSettings.filterEmail;
```
Comment (tritonJS826) [inline mw-webapp/src/logic/allUsersPage/AllUsersPage.tsx:25]: are you sure we should save filters in localstorage?

Comment (tritonJS826) [reply mw-webapp/src/logic/allUsersPage/AllUsersPage.tsx:25]: it is strange validation

**Review by tritonJS826 — APPROVED** [review]

File: mw-webapp/src/dataAccessLogic/UserPreviewDAL.ts (hunk 1 above)
Comment (Ekaterina1994) [reply mw-webapp/src/dataAccessLogic/UserPreviewDAL.ts]: Changed

File: mw-webapp/src/logic/allUsersPage/AllUsersPage.tsx (hunk 2 above)
Comment (Ekaterina1994) [reply mw-webapp/src/logic/allUsersPage/AllUsersPage.tsx]: Yea, deleted, thanks

**Pull request merged**